backward.hpp

```diff
@@ -811,13 +811,63 @@ class TraceResolverImplBase {
 
 #ifdef BACKWARD_SYSTEM_LINUX
 
+static std::string get_argv0() {
+  std::string argv0;
+  std::getline(std::ifstream("/proc/self/cmdline"), argv0, '\0');
+  return argv0;
+}
+
+static std::string read_symlink(std::string const &symlink_path) {
+  std::string path;
+  path.resize(100);
+
+  while (true) {
+    ssize_t len =
+        ::readlink(symlink_path.c_str(), &*path.begin(), path.size());
+    if (len < 0) {
+      return "";
+    }
+    if (static_cast<size_t>(len) == path.size()) {
+      path.resize(path.size() * 2);
+    } else {
+      path.resize(static_cast<std::string::size_type>(len));
+      break;
+    }
+  }
+
+  return path;
+}
+
+class TraceResolverLinuxBase
+    : public TraceResolverImplBase {
+public:
+  void fixup_dladdr_filenames(Dl_info & symbol_info, std::string & display_filename) {
```

Comment:
rename fixup_dladdr_filenames -> resolve_exec_path

return display_filename instead of using an out reference parameter.